@here/harp-mapview/lib/PickHandler.ts

```diff
@@ -147,8 +149,19 @@ export class PickHandler {
             this.mapView.textElementsRenderer.pickTextElements(scenePosition, pickResults);
         }
 
-        // calculate objects intersecting the picking ray
-        const intersects = rayCaster.intersectObjects(this.mapView.worldRootObject.children, true);
+        const intersects: THREE.Intersection[] = [];
```

Comment:
This new approach has this side effect that will not iterate objects that are not on tiles. I am not sure how much we want this, but I remember some examples adding geometry / objects as free scene entities. Would be good to ask @ninok if we would like this functionality. 
Probably the most obvious scenario is to add object via MapView anchors.

Reply:
Also previously we only iterated the tile objects (i.e. the ones in `m_worldRootObject`), so this is not changing.
Since `MapView.worldRootObject` is public for whatever reason, users could add objects there but we should forbid this and don't expose `worldRootObject`.
Nevertheless, you raised a good point, that picking is currently neither working for `mapAnchors` nor for any other object that is added to the scene.
The user could create an own raycaster with `MapView.raycasterFromScreenPoint` and intersect everything in the scene, but that would be quite slow b/c this approach would not do the early tile intersection.
I think we should have a `boolean` flag in `intersectMapObjects` to control whether only our objects or all objects are returned (even though the name `intersectMapObjects` is questionable then).
The bare minimum we should do is to test the `mapAnchor` objects as well b/c they are map objects.

Reply:
@ninok @kkostecki Another option is to insert the tiles themselves as objects in the Scene graph, and override their intersect method to continue with their children only if their intersection test succeeds.
That way we handle both tile and non-tiled objects in the same way.

Reply:
Ok good idea, let's proceed in different PR.

Reply:
Yeah, I like that idea.